tests/components/input_datetime/test_init.py

```diff
@@ -217,6 +217,34 @@ async def test_set_datetime_3(hass: HomeAssistant) -> None:
     assert state.attributes["timestamp"] == dt_obj.timestamp()
 
 
+async def test_set_datetime_4(hass: HomeAssistant) -> None:
+    """Test set_datetime method using timestamp 0."""
+    await async_setup_component(
+        hass, DOMAIN, {DOMAIN: {"test_datetime": {"has_time": True, "has_date": True}}}
+    )
+
+    entity_id = "input_datetime.test_datetime"
+
+    dt_obj = datetime.datetime(
+        1969, 12, 31, 16, 00, 00, tzinfo=dt_util.get_time_zone(hass.config.time_zone)
```

Comment:
I think we may need to freeze time to a non daylight saving date to not have a flaky test.

Reply:
Maybe not. I guess the daylight info is taken from the date created in the DateTime object here and that is fixed to December, ie a non daylight saving date.